### ISSUE TITLE
[BUG] <title> Not able to scrape Metrics of all instances

### ISSUE THREAD

**Issue by surapallykishore** [nerdswords/yet-another-cloudwatch-exporter#1418]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### YACE version

0.13.7

### Config file

config: |-
    apiVersion: v1alpha1
    discovery:
      jobs:
      - type: AWS/RDS
        regions: [ap-south-1]
        period: 300
        length: 300
        dimensionNameRequirements: [DBInstanceIdentifier]
        metrics:
          - name: CPUUtilization
            statistics: [Maximum]
          - name: DatabaseConnections
            statistics: [Sum]
          - name: FreeableMemory
            statistics: [Average]
          - name: FreeStorageSpace
            statistics: [Average]
          - name: ReadThroughput
            statistics: [Average]
          - name: WriteThroughput
            statistics: [Average]
          - name: ReadLatency
            statistics: [Maximum]
          - name: WriteLatency
            statistics: [Maximum]
          - name: ReadIOPS
            statistics: [Average]
          - name: WriteIOPS
            statistics: [Average]      
      - type: AWS/ElastiCache
        regions:
          - ap-south-1
        period: 300
        length: 300
        metrics:
          - name: CPUUtilization
            statistics: [Average]

### Current Behavior

![image](https://github.com/nerdswords/yet-another-cloudwatch-exporter/assets/140139803/b8fb65b6-6233-4770-b811-e7dd674ad51e)
![image](https://github.com/nerdswords/yet-another-cloudwatch-exporter/assets/140139803/b8fb65b6-6233-4770-b811-e7dd674ad51e)
![image](https://github.com/nerdswords/yet-another-cloudwatch-exporter/assets/140139803/b8fb65b6-6233-4770-b811-e7dd674ad51e)

### Expected Behavior

Able to see metrics of only three RDS instances while I have a total of 17 rds instances in my AWS account

**Comment by surapallykishore** [comment]
Need Help @alext

**Comment by surapallykishore** [comment]
@cristiangreco hey could you please help me on this ?

**Comment by NickAdolf** [comment]
Do the other 14 instances have tags? In the same region? Are they running\producing metrics you see in the AWS Console?

**Comment by surapallykishore** [comment]
Yes they all are in the same region and I can the see metrics on console , but YACE is scraping metrics for 3 instances .


![image](https://github.com/nerdswords/yet-another-cloudwatch-exporter/assets/140139803/b7937510-8321-4e1a-9001-d555169aa600)

**Comment by surapallykishore** [comment]
Hey @NickAdolf , I was able to see all metrics when I was using prometheus exporter but it is making tooo many API calls so switched to YACE.

**Comment by yasharne** [comment]
Same issue on Elaticache, I only get metrics of one redis cluster while we have 4 redis clusters
